pkg/plugins/handlers.go

```diff
@@ -17,32 +17,73 @@ import (
 type PluginsHandler struct {
 	Client             *http.Client
 	PluginsEndpointMap map[string]string
+	PublicDir          string
 }
 
-func NewPluginsHandler(client *http.Client, token string, pluginsEndpointMap map[string]string) *PluginsHandler {
+func NewPluginsHandler(client *http.Client, pluginsEndpointMap map[string]string, publicDir string) *PluginsHandler {
 	return &PluginsHandler{
 		Client:             client,
 		PluginsEndpointMap: pluginsEndpointMap,
+		PublicDir:          publicDir,
 	}
 }
 
-func (p *PluginsHandler) HandlePlugins(w http.ResponseWriter, r *http.Request) {
+func (p *PluginsHandler) HandlePluginLocales(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+
+	query := r.URL.Query()
+	lang := query.Get("lng")
+	// In case of static plugin the localization namespace should contain name of the json file that contains the localization strings,
+	// since various static plugins have different names for the localization file.
+	// eg. namespace query for the local-storage-operator-plugin locales - '&ns=lso-plugin.json'
+	// In case of the dynamic plugin the localization namespace should contain name of the plugin prefixed with 'plugin__' prefix.
+	// No file extension is needed since for the dynamic plugins we will fetch file from the plugin pod, based on the namespace.
+	// eg. 'plugin__helm-plugin' will fetch `helm-plugin.json`
+	namespace := query.Get("ns")
+	if !strings.HasPrefix(namespace, "plugin__") {
+		http.ServeFile(w, r, path.Join(p.PublicDir, "locales", lang, namespace))
+		return
+	}
+	// in case of dynamic-plugin we need to trim the "plugin__" prefix.
+	pluginName := strings.TrimPrefix(namespace, "plugin__")
```

Comment:
On the plus side, this would prevent e.g. `plugin__kubevirt-plugin.json` file names :smiley: 

Having i18n namespace match the file name sounds pretty straight-forward, sounds OK to me.

Reply:
@spadgett I trim the `plugin__` prefix to get the plugin service endpoint form the map of all the active plugins, where we are mapping the plugin name to its service endpoint. Based on the conversation I dont think ConsolePlugin resource name will contain the prefix, thats why Im trimming it to get the real name.